test/UnbrandedProjects.Tests/ClientUriBuilderTests.cs

```diff
@@ -0,0 +1,461 @@
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+using UnbrandedTypeSpec;
+
+namespace UnbrandedProjects.Tests
+{
+    public class ClientUriBuilderTests
+    {
+        [TestCase("http://localhost", null, "http://localhost/")]
+        [TestCase("http://localhost/", null, "http://localhost/")]
+        [TestCase("http://localhost:12345", null, "http://localhost:12345/")]
+        [TestCase("http://localhost:12345/", null, "http://localhost:12345/")]
+        [TestCase("http://localhost/with", null, "http://localhost/with")]
+        [TestCase("http://localhost/with/", null, "http://localhost/with/")]
+        [TestCase("http://localhost:12345/with", null, "http://localhost:12345/with")]
+        [TestCase("http://localhost:12345/with/", null, "http://localhost:12345/with/")]
+        [TestCase("http://localhost", "path", "http://localhost/path")]
+        [TestCase("http://localhost/", "path", "http://localhost/path")]
+        [TestCase("http://localhost:12345", "path", "http://localhost:12345/path")]
+        [TestCase("http://localhost:12345/", "path", "http://localhost:12345/path")]
+        [TestCase("http://localhost/with", "path", "http://localhost/withpath")]
+        [TestCase("http://localhost/with/", "path", "http://localhost/with/path")]
+        [TestCase("http://localhost:12345/with", "path", "http://localhost:12345/withpath")]
+        [TestCase("http://localhost:12345/with/", "path", "http://localhost:12345/with/path")]
+        [TestCase("http://localhost", "/path", "http://localhost/path")]
+        [TestCase("http://localhost/", "/path", "http://localhost/path")]
+        [TestCase("http://localhost:12345", "/path", "http://localhost:12345/path")]
+        [TestCase("http://localhost:12345/", "/path", "http://localhost:12345/path")]
+        [TestCase("http://localhost/with", "/path", "http://localhost/with/path")]
+        [TestCase("http://localhost/with/", "/path", "http://localhost/with/path")]
+        [TestCase("http://localhost:12345/with", "/path", "http://localhost:12345/with/path")]
+        [TestCase("http://localhost:12345/with/", "/path", "http://localhost:12345/with/path")]
+        [TestCase("http://localhost", "/path/", "http://localhost/path/")]
+        [TestCase("http://localhost/", "/path/", "http://localhost/path/")]
+        [TestCase("http://localhost:12345", "/path/", "http://localhost:12345/path/")]
+        [TestCase("http://localhost:12345/", "/path/", "http://localhost:12345/path/")]
+        [TestCase("http://localhost/with", "/path/", "http://localhost/with/path/")]
+        [TestCase("http://localhost/with/", "/path/", "http://localhost/with/path/")]
+        [TestCase("http://localhost:12345/with", "/path/", "http://localhost:12345/with/path/")]
+        [TestCase("http://localhost:12345/with/", "/path/", "http://localhost:12345/with/path/")]
+        public void ResetThenAppendPath(string endpoint, string pathPart, string expected)
```

Comment:
At some point in the future, we should add tests to validate that special characters like '\u1234` and "%#?&" are not escaped when we pass escape=false.  But this work is represented by https://github.com/Azure/autorest.csharp/issues/4480 and does not need to block this PR.